wasm/src/graph_traversal.rs

```diff
@@ -319,6 +334,122 @@ fn rec_flatten_traversal_data(mut data: RecTraversalData, result: &mut Vec<RecTr
     result.push(data);
 }
 
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct FlatRecTraversalData {
+    /// the edge struct that was traversed
+    #[wasm_bindgen(getter_with_clone)]
+    pub edge: EdgeStruct,
+    /// the depth of the node in the traversal
+    pub depth: u32,
+    /// the number of total children of the node, so also children of children
+    pub number_of_children: u32,
+    /// the children of the node
+    #[wasm_bindgen(getter_with_clone)]
+    pub children: Vec<usize>,
+}
+
+impl FlatRecTraversalData {
+    pub fn new(
+        edge: EdgeStruct,
+        depth: u32,
+        number_of_children: u32,
+        children: Vec<usize>,
+    ) -> FlatRecTraversalData {
+        FlatRecTraversalData {
+            edge,
+            depth,
+            number_of_children,
+            children,
+        }
+    }
+}
+
+#[wasm_bindgen]
+impl FlatRecTraversalData {
+    pub fn get_attribute_label(&self, attributes: Vec<String>) -> String {
+        self.edge.get_attribute_label(attributes)
+    }
+}
+
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct FlatRecTraversalResult {
```

Comment:
Just at a glance, can we use this same struct to handle the `flatten` method on RecTraversalData? It seems there's some duplicated logic

Reply:
it's similar, but there are some differences, though the names are too close for my liking. Maybe they can be put into one method though,